browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchActivity.java

```diff
@@ -1,44 +1,27 @@
 package com.braintreepayments.browserswitch;
 
 import android.app.Activity;
-import android.net.Uri;
 import android.os.Bundle;
 
-import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.appcompat.app.AppCompatActivity;
 
 /**
  * <a href="https://developer.android.com/guide/topics/manifest/activity-element.html#lmode">singleTask</a>
  * Activity used to receive the response from a browser switch. This Activity contains no UI and
  * finishes during {@link Activity#onCreate(Bundle)}.
  */
-public class BrowserSwitchActivity extends Activity {
+public class BrowserSwitchActivity extends AppCompatActivity {
 
-    private static Uri sReturnUri;
+    @VisibleForTesting
+    // TODO: Refactor browser-switch-android to allow injection of a custom url scheme
+    // via manifest. Ref: https://developer.android.com/studio/build/manifest-build-variables
+    BrowserSwitchClient browserSwitchClient = BrowserSwitchClient.newInstance(null);
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
-        sReturnUri = null;
-        if (getIntent() != null && getIntent().getData() != null) {
-            sReturnUri = getIntent().getData();
-        }
-
+        browserSwitchClient.captureResult(getIntent(), this);
         finish();
     }
-
-    /**
-     * @return the uri returned from the browser switch, or {@code null}.
-     */
-    @Nullable
-    public static Uri getReturnUri() {
-        return sReturnUri;
-    }
-
-    /**
-     * Clears the return uri.
-     */
-    public static void clearReturnUri() {
```

Comment:
I see we're removing a couple public methods here - isn't this a breaking change?

Reply:
Yeah these are public but the merchant shouldn't ever have to use BrowserSwitchActivity directly. Also, this logic now lives in BrowserSwitchClient.

Going forward, we may allow breaking changes for this next version.

Reply:
If we're already introducing breaking change by removing public methods like this - why do we need to keep BrowserSwitchActivity and BrowserSwitchFragment around at all anymore? I am a bit confused on how how some of this PR will be backwards compatible and some wont.

Reply:
Yeah the main backward compatible portion is BrowserSwitchFragment. That's the only part that the merchant interacts with (transitively, via BraintreeFragment).

BrowserSwitchActivity has public access, but it's technically not merchant facing. We don't have anything in the docs about BrowserSwitchActivity except for adding it to the AndroidManifest.xml to allow deep links to work.

Reply:
Yeah I agree as well. I'm up for the bump. Removing these methods cleans up the code a bit since we're no longer relying on global state for Browser Switch to work. I think that warrants a breaking change in this case and gets us to v1.0.

Reply:
@sshropshire Did we come to a conclusion on this? Holly makes a good point that devs use this SDK who aren't necessarily using Braintree. Even though these changes don't affect BT merchants, they're potential breaking changes to other devs using this SDK. Can we note these updates in the CHANGELOG as well?

Reply:
Yeah, I realize now that we would need to add Breaking Changes to the CHANGELOG I'll make sure to update those. I think we decided to just go for the breaking change here and make it a major version bump deal.